opentracing-kotlin/src/main/kotlin/org/zalando/opentracing/kotlin/TracerExtension.kt

```diff
@@ -0,0 +1,90 @@
+package org.zalando.opentracing.kotlin
+
+import io.opentracing.Scope
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.propagation.TextMapAdapter
+import io.opentracing.tag.Tags
+
+inline fun <T : Tracer, R> T.trace(span: Span, block: (Span) -> R): R {
+
+    var scope: Scope? = null
+    var closed = false
+
+    try {
+        scope = scopeManager().activate(span)
+
+        return block(span)
+    } catch (e: Exception) {
+        span.captureException(e)
+        try {
+            scope?.close()
+            closed = true
+        } catch (closeException: Exception) {
+            // Do Nothing
+        }
+        throw e
+    } finally {
+        if (!closed)
+            scope?.close()
+        span.finish()
+    }
+}
+
+inline fun <T : Tracer, R> T.trace(
+    operation: String,
+    spanBuilder: Tracer.SpanBuilder.() -> Unit,
+    block: (Span) -> R
+): R {
+    val builder = buildSpan(operation)
+    builder.apply(spanBuilder)
+    val span = builder.start()
+    return trace(span, block)
+}
+
+inline fun <T : Tracer, R> T.trace(
+    operation: String,
+    component: String? = null,
+    kind: String? = null,
+    parentSpanCtx: SpanContext? = null,
+    block: (Span) -> R
+): R {
+    return trace(
+        operation = operation,
+        spanBuilder = {
+            parentSpanCtx?.let { asChildOf(parentSpanCtx) }
+            component?.let { withTag(Tags.COMPONENT, component) }
+            kind?.let { withTag(Tags.SPAN_KIND, kind) }
+        },
+        block = block
+    )
+}
+
+fun Span.captureException(ex: java.lang.Exception) {
+    val exceptionLogs: MutableMap<String, Any> =
+        LinkedHashMap(2)
+    exceptionLogs["event"] = Tags.ERROR.key
+    exceptionLogs["error.object"] = ex
+    log(exceptionLogs)
+    Tags.ERROR[this] = true
```

Comment:
`setTag(Tags.ERROR, true)` would be significantly easier to read and understand.